app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.kt

```diff
@@ -155,6 +176,7 @@ class BrowserFragment : IWebViewLifecycleFragment(),
     override fun isBackEnabled() = canGoBack()
     override fun isForwardEnabled() = canGoForward()
     override fun getCurrentUrl() = url
+    override fun isURLPinned() = url?.let { CustomTilesAccessor.getInstance(context).isURLPinned(url!!) } ?: false
```

Comment:
Use `it`, not `url` (or bind to `url` with `?.let { url -> ... }`. If the compiler forces you to use `!!`, it means that some other thread could potentially change this value from the time you null-checked it to when you use it, which can cause bugs, so do your best not to use !! unless you're *sure* it won't change

Reply:
Good call :)